packages/__docs__/src/CodeSandboxButton/index.tsx

```diff
@@ -0,0 +1,184 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import React, { Component } from 'react'
+import { getParameters } from 'codesandbox/lib/api/define'
+
+import { Tooltip } from '@instructure/ui-tooltip'
+import { SVGIcon } from '@instructure/ui-svg-images'
+
+import { IconButton } from '@instructure/ui-buttons'
+import { propTypes, allowedProps } from './props'
+import type { CodeSandboxButtonProps } from './props'
+class CodeSandboxButton extends Component<CodeSandboxButtonProps> {
+  static propTypes = propTypes
+  static allowedProps = allowedProps
+  static defaultProps = {
+    options: {},
+    render: true
+  }
+  render() {
+    const importReact = `import React from "react"
+import ReactDOM from "react-dom"
+import moment from 'moment'
+import {avatarSquare, avatarPortrait, lorem} from "./samplemedia"
+import {placeholderImage} from "./samplemedia"
+import iconExample from "!svg-inline-loader!./heart_lg.svg"
+`
+    const pattern = /<[A-Z]\w+|Icon\w+/gm
+    const neededClasses = this.props.code
+      .match(pattern)
+      ?.map((className) => className.replace(/</gm, '').trim())
+    if (this.props.code.includes('Calendar')) {
+      neededClasses?.push('Calendar')
+    }
+    const uniqueClass = [...new Set(neededClasses)]
+    const { render } = this.props
+    const externalElements = this.props.code
+      .match(/class.\w+|function.\w+|const.\w+/gm)
+      ?.map((className) =>
+        className.replace(/class|function|const/gm, '').trim()
+      )
+    const allNeededClasses = uniqueClass?.filter(
+      (className) => !externalElements?.includes(className)
+    )
+    const importClasses = `import {${allNeededClasses.join(
+      ', '
+    )}} from "@instructure/ui"\n\n`
+    const code = render ? `render(${this.props.code})` : this.props.code
+    const renderStatement = `const render = (el) => { ReactDOM.render(el, document.getElementById('app')) }\n`
+    const data = {
+      title: this.props.title,
+      js: importReact + importClasses + renderStatement + code,
+      private: true,
+      editors: '001',
+      html: '<div id="app"></div><div id="flash-messages"></div><div id="nav"></div>',
+      css_prefix: 'autoprefixer',
+      js_pre_processor: 'babel',
+      ...this.props.options
+    }
+    const dependencies = JSON.stringify({
+      dependencies: {
+        '@instructure/debounce': '^8',
+        '@instructure/ui': '^8',
+        '@instructure/ui-icons': '^8',
+        'lorem-ipsum': '^1.0.0',
+        react: '17.0.2',
+        'react-dom': '17.0.2',
+        'react-scripts': '4.0.0',
+        moment: '^2.23.0',
+        'svg-inline-loader': '0.8.2'
+      }
+    }).replace(/'/g, '&apos;')
```

Comment:
if you add here the `space` parameter too the `package.json` will be nicely formatted:

```
}, null, 2).replace(/'/g, '&apos;')
```

Reply:
Done